### PR TITLE
Fix scala==2.13 nix build, cleanup

### PR DESCRIPTION
**Pull request by jonringer**
# Description

Fix build with scala 2.13

cc @input-output-hk/mantis-devs

### CODE REVIEW

**Review by dmitry-worker — APPROVED** [review]
Thanks @jonringer! We now have
 - no more repo.nix
 - no sbtix-related stuff in code

That's awesome! :+1: 

But there should be a fix for buildkite, cause it tries to follow the old build scenario.
Would you like me to update it or maybe we can manage it together using tmate? 
It would be great if one can test buildkite without commiting anything to the repo...

Regarding this PR - I believe I'm not an expert in Nix packaging, so we may require @manveru 's review.

File: nix/pkgs/mantis.nix [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  nativeBuildInputs = [ protoc-wrapper jdk makeWrapper ];
+
+  # read version from build.sbt
+  version = let
+    buildSbt = builtins.readFile ../../build.sbt;
+    captures = builtins.match ''.*version := "([^"]+)".*'' buildSbt;
+  in builtins.elemAt captures 0;
+
+  LD_LIBRARY_PATH = lib.makeLibraryPath [ stdenv.cc.cc.lib ];
+  PATH = lib.makeBinPath [ jre gawk ];
```
Comment (manveru) [inline nix/pkgs/mantis.nix]: Please add coreutils to `PATH`. It's required by the launcher scripts

**Review by manveru — CHANGES_REQUESTED** [review]
looks good otherwise, just needs one change.

File: nix/pkgs/mantis.nix (hunk 1 above)
Comment (jonringer) [reply nix/pkgs/mantis.nix]: it's why I'm appending to the previous path, but I think your way is probably more correct, thanks :)

Comment (jonringer) [reply nix/pkgs/mantis.nix]: fixed

**Pull request merged**